c7n/resources/ebs.py

```diff
@@ -446,6 +449,57 @@ def process(self, resources, event=None):
         return [r for r in resources if r['VolumeId'] in flagged]
 
 
+@filters.register('health-event')
+class HealthFilter(HealthEventFilter):
+
+    def process(self, resources, event=None):
+        if not resources:
+            return resources
+
+        client = local_session(self.manager.session_factory).client('health')
+        paginator = client.get_paginator('describe_events')
+        m = self.manager
+        result_map = {}
+
+        f = {'services': [m.data['resource'].upper()],
+             'eventStatusCodes': self.data.get(
+                 'statuses', ['open', 'upcoming'])}
+
+        if self.data.get('types'):
+            f['eventTypeCodes'] = self.data.get('types')
+
+        events = list(itertools.chain(
+            *[p['events']for p in paginator.paginate(filter=f)]))
+
+        for event_set in chunks(events, 10):
+            event_map = {e['arn']: e for e in event_set}
+            for d in client.describe_event_details(
+                    eventArns=event_map.keys()).get('successfulSet', ()):
+                event_map[d['event']['arn']]['Description'] = d[
+                    'eventDescription']['latestDescription']
+            entities = client.describe_affected_entities(
+                filter={'eventArns': event_map.keys()})['entities']
+
+            for e in entities:
+                rid = e['entityValue']
+                if not result_map.get(rid):
+                    result_map[rid] = self.load_resource(rid)
+
+                result_map[rid].setdefault(
+                    'c7n:HealthEvent', []).append(event_map[e['eventArn']])
+        return result_map.values()
+
+    def load_resource(self, rid):
+        config = local_session(self.manager.session_factory).client('config')
+        paginator = config.get_paginator('get_resource_config_history')
+        for p in paginator.paginate(
```

Comment:
configuration is null in the deleted resource case and we should probably do limit=2 to handle that case, as the lost volume scenario will effectively always have a deleted resource in index 0 when doing default reverse date sort. we should be walking through the event pair retrieved to a non deleted one.